hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieConcatHandle.java

```diff
@@ -94,4 +104,22 @@ public void write(GenericRecord oldRecord) {
     }
     recordsWritten++;
   }
+
+  @Override
+  public List<WriteStatus> close() {
+    try {
+      while (recordItr.hasNext()) {
+        HoodieRecord<T> record = recordItr.next();
+        if (needsUpdateLocation()) {
+          record.unseal();
+          record.setNewLocation(new HoodieRecordLocation(instantTime, fileId));
+          record.seal();
+        }
+        writeInsertRecord(record);
+      }
+    } catch (IOException e) {
+      throw new HoodieUpsertException("Failed to close UpdateHandle", e);
+    }
+    return super.close();
```

Comment:
let's not call super.close(). lets add a protected method in MergeHandle just for processing records within close. and here awe can just over-ride that method alone. I am talking about lines 364 to 367 in MergeHandle.